sys/include/net/gnrc/tcp.h

```diff
@@ -34,6 +34,39 @@
 extern "C" {
 #endif
 
+
+/**
+ * @brief Address information for a single TCP connection endpoint.
+ * @note Must be compatible to sock_tcp_ep_t.
```

Comment:
I think this note is only of interest to you for the moment won't be that important down the road. Alternatively, I would rather go for 
```
@extends sock_tcp_ep_t
```

to make the relationship clear.